q2_moshpit/eggnog/tests/test_utils.py

```diff
@@ -0,0 +1,134 @@
+    _merge_hmms_and_write_idmap, COMMON_URL, _validate_eggnog_hmmer_taxon_id
+)
+from q2_moshpit.eggnog._format import EggnogHmmerIdmapDirectoryFmt
+from q2_types.genome_data import ProteinsDirectoryFormat
+from q2_types.profile_hmms import (
+    PressedProfileHmmsDirectoryFmt, ProteinMultipleProfileHmmDirectoryFmt
+)
+
+
+class TestEggnogUtils(TestPluginBase):
+    package = "q2_moshpit.eggnog.tests"
+
+    @patch("urllib.request.urlopen")
+    def test_validate_eggnog_hmmer_taxon_id_valid(self, mock_urlopen):
+        # Mock HTML content
+        mock_html = """
+        <html>
+            <body>
+                <a href="1/">1/</a>
+                <a href="2/">2/</a>
+                <a href="3/">3/</a>
+            </body>
+        </html>
+        """
+        mock_response = MagicMock()
+        mock_response.read.return_value = mock_html.encode('utf-8')
+        mock_urlopen.return_value.__enter__.return_value = mock_response
+
+        _validate_eggnog_hmmer_taxon_id(2)
+
+    @patch("urllib.request.urlopen")
+    def test_validate_eggnog_hmmer_taxon_id_invalid(self, mock_urlopen):
+        # Mock HTML content
+        mock_html = """
+        <html>
+            <body>
+                <a href="1/">1/</a>
+                <a href="2/">2/</a>
+                <a href="3/">3/</a>
+            </body>
+        </html>
+        """
+        mock_response = MagicMock()
+        mock_response.read.return_value = mock_html.encode('utf-8')
+        mock_urlopen.return_value.__enter__.return_value = mock_response
+
+        with self.assertRaisesRegex(
+            ValidationError, "4 is not a valid taxon ID."
+        ):
+            _validate_eggnog_hmmer_taxon_id(4)
+
+    @patch("subprocess.run")
+    @patch("tempfile.TemporaryDirectory")
+    def test_download_and_build_hmm_db(self, tmpdir, mock_run):
+        tmp = self.get_data_path('hmmer/hmms')
+        taxon_id = 1
+        tmpdir.return_value.__enter__.return_value = tmp
+
+        idmap, hmm_db, pressed_hmm_db = _download_and_build_hmm_db(taxon_id)
+
+        self.assertIsInstance(idmap, EggnogHmmerIdmapDirectoryFmt)
+        self.assertIsInstance(hmm_db, ProteinMultipleProfileHmmDirectoryFmt)
+        self.assertIsInstance(pressed_hmm_db, PressedProfileHmmsDirectoryFmt)
+
+        mock_run.assert_has_calls([
+            call(
+                [
+                    "wget", "-O", f"{tmp}/{taxon_id}_hmms.tar.gz",
+                    f"{COMMON_URL}/{taxon_id}/{taxon_id}_hmms.tar.gz"
+                ],
+                check=True
+            ),
+            call(
+                ["tar", "zxf", f"{taxon_id}_hmms.tar.gz"],
+                check=True,
+                cwd=tmp
+            ),
+            call(
+                ["hmmpress", f"{str(pressed_hmm_db)}/{taxon_id}.hmm"],
+                check=True
+            )
+        ])
+
+    @patch("subprocess.run")
+    @patch("tempfile.TemporaryDirectory")
+    def test_download_fastas_into_hmmer_db(self, tmpdir, mock_run):
```

Comment:
This test should probably also has some assertion(s) about the _contents_ of the created outputs. Otherwise, you really are only asserting that the mocked functions were called properly and that _some_ ProteinsDirFmt object was returned...